apps/solidity-compiler/src/app/compiler-api.ts

```diff
@@ -0,0 +1,259 @@
+import { compile } from '@remix-project/remix-solidity'
+import { CompileTabLogic, parseContracts } from '@remix-ui/solidity-compiler' // eslint-disable-line
+import { ConfigurationSettings } from '@remix-project/remix-lib-ts'
+
+export const CompilerApiMixin = (Base) => class extends Base {
+  currentFile: string
+  contractMap: {
+    file: string
+  } | Record<string, any>
+  compileErrors: any
+  compileTabLogic: CompileTabLogic
+  contractsDetails: Record<string, any>
+
+  configurationSettings: ConfigurationSettings
+
+  onCurrentFileChanged: (fileName: string) => void
+  onResetResults: () => void
+  onSetWorkspace: (workspace: any) => void
+  onNoFileSelected: () => void
+  onCompilationFinished: (contractsDetails: any, contractMap: any) => void
+  onSessionSwitched: () => void
+  onContentChanged: () => void
+
+  initCompilerApi () {
+    this.configurationSettings = null
+
+    this._view = {
+      warnCompilationSlow: null,
+      errorContainer: null,
+      contractEl: null
+    }
+    
+    this.contractsDetails = {}
+    this.data = {
+      eventHandlers: {},
+      loading: false
+    }
+
+    this.contractMap = {}
+    this.contractsDetails = {}
+
+    this.compileErrors = {}
+    this.compiledFileName = ''
+    this.currentFile = ''
+  }
+
+  onActivation () {
+    this.listenToEvents()    
+  }
+
+  onDeactivation () {}
+
+  resolveContentAndSave (url) {
+    return this.call('contentImport', 'resolveAndSave', url)
```

Comment:
Check if `await` should be there

Reply:
this returns a Promise,  `resolveContentAndSave` is declared with the following interface: `resolveContentAndSave: (url: string) => Promise<string>`